src/Integration/Connection/UI/OrganizationSelectionWindow.xaml

```diff
@@ -41,46 +42,98 @@
+                                </Style>
+                            </ComboBox.ItemContainerStyle>
+                            <ComboBox.ItemTemplate>
+                                <DataTemplate DataType="{x:Type models:SonarQubeOrganization}">
+                                    <TextBlock TextTrimming="CharacterEllipsis">
+                        <Run Text="{Binding Name, Mode=OneWay}" />
+                        <Run Text="(" /><Run Text="{Binding Key, Mode=OneWay}" /><Run Text=")" />
+                                    </TextBlock>
+                                </DataTemplate>
+                            </ComboBox.ItemTemplate>
+                        </ComboBox>
+                    </StackPanel>
+
+                    <Separator Style="{StaticResource DialogSeparator}" />
+
+                    <StackPanel VerticalAlignment="Center" Style="{StaticResource DialogButtonsPanel}">
+                        <vsui:DialogButton Click="OnOwnOkButtonClick" Content="OK" IsDefault="True"
+                                IsEnabled="{Binding ElementName=OrganizationComboBox,
+                                                    Path=SelectedItem,
+                                                    Converter={StaticResource NotNullConverter}}" />
+                        <vsui:DialogButton Content="Cancel" IsCancel="True" />
+                    </StackPanel>
+                </StackPanel>
+
+            </Grid>
+        </TabItem>
+        <TabItem Header="Other organizations">
+            <Grid>
+                <Grid.RowDefinitions>
+                    <RowDefinition SharedSizeGroup="tabItemSize" />
+                </Grid.RowDefinitions>
+                <Grid.ColumnDefinitions>
+                    <ColumnDefinition SharedSizeGroup="tabItemHeight" />
+                </Grid.ColumnDefinitions>
+
+                <StackPanel Style="{StaticResource DialogContent}" >
+                    <TextBlock Margin="0,0,0,8"
+                    FontStyle="Italic" Text="Please type the key of the organization you want to bind to." />
+
+                    <StackPanel Orientation="Horizontal">
+                        <TextBlock VerticalAlignment="Center" Text="Organization key" />
+                        <TextBox x:Name="OrganizationKeyTextBox"
+                            Width="250" Margin="12,0,0,0" >
+                        </TextBox>
+                    </StackPanel>
+
+                    <Separator Style="{StaticResource DialogSeparator}" />
+
+                    <StackPanel VerticalAlignment="Center" Style="{StaticResource DialogButtonsPanel}">
+                        <vsui:DialogButton Click="OnOtherOrgOkButtonClick" Content="OK" IsDefault="True"
+                                IsEnabled="{Binding ElementName=OrganizationKeyTextBox,
+                                                    Path=Text,
+                                                    Converter={StaticResource IsValidOrgKeyConverter}}" />
```

Comment:
The `IsEnabled="{Binding... Converter={StaticResource IsValidOrgKeyConverter}}` is the WPF magic that takes care of enabling/disabling the `OK` button.

WPF will take care of calling the `IsValidOrgKeyConverter` each time the contents of the textbox containing the organization key change, and enable/disable the button based on the value returned by the converter.